src/android/TrackierCordovaPlugin.java

```diff
@@ -0,0 +1,128 @@
+package com.trackier.cordova_sdk;
+
+import org.apache.cordova.CordovaPlugin;
+import org.apache.cordova.CallbackContext;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import android.util.Log;
+
+/**
+ * This class echoes a string called from JavaScript.
+ */
+public class TrackierCordovaPlugin extends CordovaPlugin {
+
+    @Override
+    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
+        if (action.equals("initializeSDK")) {
+            String message = args.getString(0);
+            this.initializeSDK(message, callbackContext);
+            return true;
+        } else if (action.equals("trackEvent")) {
+            String message = args.getString(0);
+            this.trackEvent(message, callbackContext);
+            return true;
+        }
+        return false;
+    }
+
+    private void initializeSDK(String message, CallbackContext callbackContext) throws JSONException {
```

Comment:
our function should not throw any exception they should catch it and return error